### PR TITLE
EA: tighten TimedeltaArray._from_sequence signature

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

xref #36718

### CODE REVIEW

File: pandas/core/arrays/timedeltas.py [hunk 1]
```diff
@@ -214,6 +214,19 @@ def _simple_new(
...
+            _validate_td64_dtype(dtype)
+
+        data, inferred_freq = sequence_to_td64ns(data, copy=copy, unit=None)
+        freq, _ = dtl.validate_inferred_freq(None, inferred_freq, False)
+
+        result = cls._simple_new(data, freq=freq)
+        return result
+
+    @classmethod
+    def _from_sequence_not_strict(
```
Comment (jreback) [inline pandas/core/arrays/timedeltas.py:229]: umm, have we standarized on this naming? or even the concept of 2 from_sequence?
cc @jorisvandenbossche @TomAugspurger 

IIRC we had a need for a `coerce=` keyword that we nixed for StringArray

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/timedeltas.py (hunk 1 above)
Comment (jreback) [reply pandas/core/arrays/timedeltas.py:229]: xref https://github.com/pandas-dev/pandas/pull/36718 as well

Comment (jbrockmendel) [reply pandas/core/arrays/timedeltas.py:229]: yah my best guess is we're gonna end up dropping this method altogether and having it only reached via the DTI/TDI constructors

Comment (jreback) [reply pandas/core/arrays/timedeltas.py:229]: ok as a temporary work-around prob ok.

Comment (jbrockmendel) [reply pandas/core/arrays/timedeltas.py:229]: yah, there's also more work to do in tightening what dtypes _from_sequence accepts, plenty left to do

**Review by jreback — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
I am not sure this actually makes sense. Based on the discussion in #36718 (but we should first futher discuss that), we would rather make `_from_sequence` the non-strict version, so the other way around as you did here (now, it's only naming, so easy to change later of course)

**Comment by jbrockmendel** [comment]
@jorisvandenbossche there isn't a whole lot of discussion in #36718.  did you mean #33254?

**Comment by jorisvandenbossche** [comment]
> there isn't a whole lot of discussion in #36718. did you mean #33254?

Yes, indeed, sorry for the typo ;) 
Will take a look at your comment over there now!

**Pull request merged**